src/main/java/co/rsk/bitcoinj/core/StoredBlock.java

```diff
@@ -113,13 +120,33 @@ public StoredBlock getPrev(BtcBlockStore store) throws BlockStoreException {
         return store.get(getHeader().getPrevBlockHash());
     }
 
-    /** Serializes the stored block to a custom packed format. Used by {@link CheckpointManager}. */
-    public void serializeCompact(ByteBuffer buffer) {
-        byte[] chainWorkBytes = Utils.bigIntegerToBytes(getChainWork(), CHAIN_WORK_BYTES);
-        if (chainWorkBytes.length < CHAIN_WORK_BYTES) {
-            // Pad to the right size.
-            buffer.put(EMPTY_BYTES, 0, CHAIN_WORK_BYTES - chainWorkBytes.length);
-        }
+    /**
+     * @deprecated Use {@link #serializeCompactV2(ByteBuffer)} instead.
+     *
+     * Serializes the stored block to a custom packed format. Used internally.
+     * As of June 22, 2024, it takes 12 unsigned bytes to store the chain work value,
+     * so developers should use the V2 format.
+     *
+     * @param buffer buffer to write to
+     */
+    @Deprecated
+    public void serializeCompactLegacy(ByteBuffer buffer) {
```

Comment:
Looks like both methods have the same identical implementation except for the chain work size right?

If that's the case perhaps we could move the logic to a private method that receives the chain work size as param

Reply:
Updated